framework/PostProcessors/DataClassifier.py

```diff
@@ -0,0 +1,318 @@
+          newInput[dataType]['input'] = dict.fromkeys(inputObject.getParaKeys('inputs'), None)
+          newInput[dataType]['output'] = dict.fromkeys(inputObject.getParaKeys('outputs'), None)
+          newInput[dataType]['type'] = inputObject.type
+          newInput[dataType]['name'] = inputObject.name
+
+          rlzKeys = inputObject.getOutParametersValues.keys()
+          newInput[dataType]['historySizes'] = dict.fromkeys(rlzKeys, None)
+          for rlzIndex, rlzKey in enumerate(rlzKeys):
+            realization = inputObject.getRealization(rlzIndex)
+            for param in inputObject.getParaKeys('inputs'):
+              if newInput[dataType]['input'][param] is None:
+                newInput[dataType]['input'][param] = np.empty(0)
+              newInput[dataType]['input'][param] = np.append(newInput[dataType]['input'][param], realization['inputs'][param][-1])
+            for param in inputObject.getParaKeys('outputs'):
+              if newInput[dataType]['output'][param] is None:
+                newInput[dataType]['output'][param] = np.empty(0)
+              newInput[dataType]['output'][param] = np.append(newInput[dataType]['output'][param], realization['outputs'][param][-1])
+              if newInput[dataType]['historySizes'][rlzKey] is None:
+                newInput[dataType]['historySizes'][rlzKey] = len(realization['outputs'][param])
+
+    return newInput
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action.
+      @ In,  inputIn, list, list of DataObjects
+      @ Out, None
+    """
+    inputDict = self.inputToInternal(inputIn)
+    targetDict = inputDict['target']
+    classifierDict = inputDict['classifier']
+    outputDict = {}
+    outputType = targetDict['type']
+    outputDict['dataType'] = outputType
+    outputDict['dataFrom'] = targetDict['name']
+    if outputType == 'HistorySet':
+      outputDict['historySizes'] = copy.copy(targetDict['historySizes'])
+
+    outputDict[self.label] = np.empty(0)
+    numRlz = targetDict['input'].values()[0].size
+    for i in range(numRlz):
+      tempTargDict = {}
+      for param, vals in targetDict['input'].items():
+        tempTargDict[param] = vals[i]
+      for param, vals in targetDict['output'].items():
+        tempTargDict[param] = vals[i]
+      tempClfList = []
+      labelIndex = None
+      for key, values in classifierDict['input'].items():
+        calcVal = self.funcDict[key].evaluate("evaluate", tempTargDict)
+        inds, = np.where(values == calcVal)
+        if labelIndex is None:
+           labelIndex = set(inds)
+        else:
+          labelIndex = labelIndex & set(inds)
+      if len(labelIndex) != 1:
+        self.raiseAnError(IOError, "The parameters", ",".join(tempTargDict.keys()), "with values", ",".join([str(el) for el in tempTargDict.values()]), "could not be classifier!")
+      outputDict[self.label] = np.append(outputDict[self.label], classifierDict['output'][self.label][list(labelIndex)[0]])
```

Comment:
would it make sense to avoid the append on a np array since we know already the size of outputDict[self.label]?

Reply:
fixed.